jormungandr/src/explorer/graphql/mod.rs

```diff
@@ -63,16 +64,54 @@ impl Block {
     }
 
     /// The transactions contained in the block
-    pub fn transactions(&self, context: &Context) -> FieldResult<Vec<Transaction>> {
-        Ok(self
-            .get_explorer_block(&context.db)?
-            .transactions
-            .iter()
-            .map(|(id, _tx)| Transaction {
-                id: id.clone(),
-                in_block: self.hash.clone(),
-            })
-            .collect())
+    pub fn transactions(
+        &self,
+        first: Option<i32>,
+        last: Option<i32>,
+        before: Option<IndexCursor>,
+        after: Option<IndexCursor>,
+        context: &Context,
+    ) -> FieldResult<TransactionConnection> {
+        let explorer_block = self.get_explorer_block(&context.db)?;
+        let mut transactions: Vec<&ExplorerTransaction> =
+            explorer_block.transactions.values().collect();
+
+        // TODO: This may be expensive at some point, but I can't rely in
+        // the HashMap's order (also, I'm assuming the order in the block matters)
+        transactions
+            .as_mut_slice()
+            .sort_unstable_by_key(|tx| tx.offset_in_block);
+
+        let lower_bound = 0u32;
+        let upper_bound = transactions.len().checked_sub(1).or(Some(0)).unwrap();
+
+        TransactionConnection::new(
+            lower_bound,
+            upper_bound
+                .try_into()
+                .expect("less than 2^32 transactions per block"),
+            first,
+            last,
+            before,
+            after,
+            |from: u32, to: u32| {
+                use std::cmp::min;
+                transactions[usize::try_from(from).unwrap()
+                    ..min(
+                        transactions.len(),
+                        usize::try_from(to.checked_add(1).unwrap()).unwrap(),
```

Comment:
It would be better for readability if these long sanitization expressions were on separate lines as `let from = ...`.